### PR TITLE
🐛 Bugfix: Fixed an issue where the MCP service failed to start in a Kubernetes pod.

### LINKED ISSUE #3253: [Bug] 默认租户下k8s容器化启动MCP服务失败，因为容器名称中不能包含_
### Nexent Version

_No response_

### Problem Description

默认租户下k8s容器化启动MCP服务失败，因为容器名称中不能包含_

### Reproduction Steps

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by YehongPan**
🐛 Bugfix: Fixed an issue where the MCP service failed to start in a Kubernetes pod.
fixes #3253 
[Specification Details]
1. Modify the pod naming logic to convert all non-compliant characters to -.
2. Modify test cases.

### CODE REVIEW

File: sdk/nexent/container/k8s_client.py [hunk 1]
```diff
@@ -23,6 +24,47 @@
 
 logger = logging.getLogger("nexent.container.kubernetes")
 
+# Kubernetes naming constraints: lowercase alphanumeric or dash, cannot start/end with dash,
+# cannot have consecutive dashes, max 253 characters
```
Comment (YehongPan) [inline sdk/nexent/container/k8s_client.py:28]: **[逻辑漏洞]** `_sanitize_k8s_name` 函数在注释中提到了 Kubernetes 命名约束（最大 253 字符），但实际代码中未对长度进行截断。如果输入名称超过 253 字符，生成的 K8s 资源名称将无效，导致创建失败。建议在函数末尾添加长度截断逻辑：`sanitized = sanitized[:253].rstrip('-')`。

File: sdk/nexent/container/k8s_client.py [hunk 2]
```diff
@@ -23,6 +24,47 @@
...
+K8S_NAME_PATTERN = re.compile(r"[^a-z0-9-]+")
+K8S_CONSECUTIVE_DASHES = re.compile(r"-+")
+
+
+def _sanitize_k8s_name(name: str) -> str:
+    """Convert arbitrary string to valid Kubernetes resource name.
+
+    Rules:
+    - Convert to lowercase
```
Comment (JasonW404) [inline sdk/nexent/container/k8s_client.py:37]: `_sanitize_k8s_name(None)` 会在 `name.lower()` 处抛出 `AttributeError`，因为 `None` 没有 `lower()` 方法。`if not name` 检查在 `name.lower()` 之后才执行，无法拦截 None 输入。测试用例 `test_sanitize_none` 预期返回 "unknown" 但实际会崩溃。建议将 None 检查移到最前面：`if not name: return "unknown"` 改为 `if name is None or not name: return "unknown"`。

**Review by WMC001 — COMMENTED** [review]
**Observation: improved K8s name sanitization**

`sdk/nexent/container/k8s_client.py` — extracts `_sanitize_k8s_name` helper for centralized name sanitization across `_generate_pod_name` and `list_containers`. The helper handles empty/None inputs (returns `"unknown"`), strips leading/trailing dashes, and prepends `x` for names that start with non-alphanumeric. Tests cover the main edge cases including consecutive special chars, leading/trailing chars, and underscores/dots/spaces.

No bugs found. Note that `_sanitize_k8s_name` is called before the `[:8]` truncation, which is the correct order — internal `.strip("-")` ensures the truncated slice doesn't contain leading/trailing dashes.

File: sdk/nexent/container/k8s_client.py [hunk 3]
```diff
@@ -23,6 +24,47 @@
...
+
+    # Remove leading/trailing dashes
+    sanitized = sanitized.strip("-")
+
+    # Ensure it starts with alphanumeric
+    if sanitized and not sanitized[0].isalnum():
+        sanitized = "x" + sanitized
+
+    # Fallback if empty
+    return sanitized if sanitized else "unknown"
```
Comment (xuyaqist) [inline sdk/nexent/container/k8s_client.py:66]: `_sanitize_k8s_name` 函数没有对输出长度进行限制。Kubernetes DNS label 名称最长 63 个字符（RFC 1123），而 pod 名称由 `mcp-{safe_name}-{tenant}-{user}-{uuid}` 拼接而成。如果 `service_name` 很长，最终 pod 名称可能超过 253 字符限制导致 K8s API 报错。建议在函数末尾添加截断逻辑，或在 `_generate_pod_name` 中对各段长度做约束。

File: sdk/nexent/container/k8s_client.py [hunk 4]
```diff
@@ -23,6 +24,47 @@
 
 logger = logging.getLogger("nexent.container.kubernetes")
 
+# Kubernetes naming constraints: lowercase alphanumeric or dash, cannot start/end with dash,
+# cannot have consecutive dashes, max 253 characters
+K8S_NAME_PATTERN = re.compile(r"[^a-z0-9-]+")
+K8S_CONSECUTIVE_DASHES = re.compile(r"-+")
+
+
+def _sanitize_k8s_name(name: str) -> str:
```
Comment (xuyaqist) [inline sdk/nexent/container/k8s_client.py:33]: `_sanitize_k8s_name` 在 `sanitized` 为空时返回 `"unknown"`，但当输入为 `None` 时，虽然 `if not name` 的 guard 能正确拦截（返回 True），但类型注解 `name: str` 应改为 `Optional[str]` 以准确反映函数实际接受的输入类型，提高代码可读性和类型检查的准确性。

File: sdk/nexent/container/k8s_client.py (hunk 1 above)
Comment (hhhhsc701) [inline sdk/nexent/container/k8s_client.py:28]: [P2] 注释写 max 253，但这个 sanitizer 没有实际 enforce 长度，而且很多使用点是 label value/名称片段，常见上限是 63。注释会让调用方误以为长度已经安全，建议把长度作为参数并在函数内截断。

File: sdk/nexent/container/k8s_client.py [hunk 5]
```diff
@@ -23,6 +24,47 @@
...
+        name: Input string to sanitize
+
+    Returns:
+        Valid Kubernetes name (lowercase alphanumeric and dashes only)
+    """
+    if not name:
+        return "unknown"
+
+    # Lowercase and replace invalid chars with dash
+    sanitized = K8S_NAME_PATTERN.sub("-", name.lower())
```
Comment (hhhhsc701) [inline sdk/nexent/container/k8s_client.py:53]: [P2] name 标注为 str，但测试又传 None；如果传入 truthy 的非字符串对象，name.lower() 会抛 AttributeError。边界入口应先 str(name) 或显式拒绝非字符串。

File: sdk/nexent/container/k8s_client.py (hunk 3 above)
Comment (hhhhsc701) [inline sdk/nexent/container/k8s_client.py:66]: [P1] 返回值没有任何长度限制，长 service_name 经过清洗后仍然可以非常长，最终 pod name 会超过 Kubernetes 限制。这里应该在 sanitizer 或 _generate_pod_name 中截断并附稳定 hash。

File: sdk/nexent/container/k8s_client.py [hunk 6]
```diff
@@ -77,9 +119,9 @@ def __init__(self, config: KubernetesContainerConfig):
 
     def _generate_pod_name(self, service_name: str, tenant_id: str, user_id: str) -> str:
         """Generate unique pod name with service, tenant, and user segments."""
-        safe_name = "".join(c if c.isalnum() or c == "-" else "-" for c in service_name)
-        tenant_part = (tenant_id or "")[:8]
-        user_part = (user_id or "")[:8]
+        safe_name = _sanitize_k8s_name(service_name)
+        tenant_part = _sanitize_k8s_name(tenant_id)[:8]
```
Comment (hhhhsc701) [inline sdk/nexent/container/k8s_client.py:123]: [P2] tenant_part/user_part 只取前 8 位，tenantabcdef 和 tenantabcxyz 会生成同样前缀，排障和清理时容易误判。建议用前缀+短 hash，而不是纯截断。

File: sdk/nexent/container/k8s_client.py [hunk 7]
```diff
@@ -486,7 +528,7 @@ def list_containers(
 
                 # Filter by service_name if provided
                 if service_name:
-                    safe_name = "".join(c if c.isalnum() or c == "-" else "-" for c in service_name)
+                    safe_name = _sanitize_k8s_name(service_name)
```
Comment (hhhhsc701) [inline sdk/nexent/container/k8s_client.py:531]: [P2] list_containers 用 sanitize 后的 safe_name 再做子串匹配，service=api 会匹配到 my-api-v2 等不相关 pod。应和创建时的 component label 做精确匹配，或保存完整 sanitized service_name label。

File: sdk/nexent/container/k8s_client.py (hunk 1 above)
Comment (hhhhsc701) [inline sdk/nexent/container/k8s_client.py:28]: [P2] 注释写 max 253，但这个 sanitizer 没有实际 enforce 长度，而且很多使用点是 label value/名称片段，常见上限是 63。注释会让调用方误以为长度已经安全，建议把长度作为参数并在函数内截断。

File: sdk/nexent/container/k8s_client.py (hunk 5 above)
Comment (hhhhsc701) [inline sdk/nexent/container/k8s_client.py:53]: [P2] name 标注为 str，但测试又传 None；如果传入 truthy 的非字符串对象，name.lower() 会抛 AttributeError。边界入口应先 str(name) 或显式拒绝非字符串。

File: sdk/nexent/container/k8s_client.py (hunk 3 above)
Comment (hhhhsc701) [inline sdk/nexent/container/k8s_client.py:66]: [P1] 返回值没有任何长度限制，长 service_name 经过清洗后仍然可以非常长，最终 pod name 会超过 Kubernetes 限制。这里应该在 sanitizer 或 _generate_pod_name 中截断并附稳定 hash。

File: sdk/nexent/container/k8s_client.py (hunk 6 above)
Comment (hhhhsc701) [inline sdk/nexent/container/k8s_client.py:123]: [P2] tenant_part/user_part 只取前 8 位，tenantabcdef 和 tenantabcxyz 会生成同样前缀，排障和清理时容易误判。建议用前缀+短 hash，而不是纯截断。

File: sdk/nexent/container/k8s_client.py (hunk 7 above)
Comment (hhhhsc701) [inline sdk/nexent/container/k8s_client.py:531]: [P2] list_containers 用 sanitize 后的 safe_name 再做子串匹配，service=api 会匹配到 my-api-v2 等不相关 pod。应和创建时的 component label 做精确匹配，或保存完整 sanitized service_name label。

File: sdk/nexent/container/k8s_client.py (hunk 1 above)
Comment (Dallas98) [inline sdk/nexent/container/k8s_client.py:28]: 事后审查补充：[P2] 注释写 max 253，但这个 sanitizer 没有实际 enforce 长度，而且很多使用点是 label value/名称片段，常见上限是 63。注释会让调用方误以为长度已经安全，建议把长度作为参数并在函数内截断。

影响：这个问题合入后会在对应部署、运行或权限场景中留下真实故障/安全风险，后续排查成本较高。
建议：沿着上述风险点补齐校验、配置来源、权限边界或回归测试，避免同类问题再次出现。

File: sdk/nexent/container/k8s_client.py (hunk 5 above)
Comment (Dallas98) [inline sdk/nexent/container/k8s_client.py:53]: 事后审查补充：[P2] name 标注为 str，但测试又传 None；如果传入 truthy 的非字符串对象，name.lower() 会抛 AttributeError。边界入口应先 str(name) 或显式拒绝非字符串。

影响：这个问题合入后会在对应部署、运行或权限场景中留下真实故障/安全风险，后续排查成本较高。
建议：沿着上述风险点补齐校验、配置来源、权限边界或回归测试，避免同类问题再次出现。

File: sdk/nexent/container/k8s_client.py (hunk 3 above)
Comment (Dallas98) [inline sdk/nexent/container/k8s_client.py:66]: 事后审查补充：[P1] 返回值没有任何长度限制，长 service_name 经过清洗后仍然可以非常长，最终 pod name 会超过 Kubernetes 限制。这里应该在 sanitizer 或 _generate_pod_name 中截断并附稳定 hash。

影响：这个问题合入后会在对应部署、运行或权限场景中留下真实故障/安全风险，后续排查成本较高。
建议：沿着上述风险点补齐校验、配置来源、权限边界或回归测试，避免同类问题再次出现。

File: sdk/nexent/container/k8s_client.py (hunk 6 above)
Comment (Dallas98) [inline sdk/nexent/container/k8s_client.py:123]: 事后审查补充：[P2] tenant_part/user_part 只取前 8 位，tenantabcdef 和 tenantabcxyz 会生成同样前缀，排障和清理时容易误判。建议用前缀+短 hash，而不是纯截断。

影响：这个问题合入后会在对应部署、运行或权限场景中留下真实故障/安全风险，后续排查成本较高。
建议：沿着上述风险点补齐校验、配置来源、权限边界或回归测试，避免同类问题再次出现。

File: sdk/nexent/container/k8s_client.py (hunk 7 above)
Comment (Dallas98) [inline sdk/nexent/container/k8s_client.py:531]: 事后审查补充：[P2] list_containers 用 sanitize 后的 safe_name 再做子串匹配，service=api 会匹配到 my-api-v2 等不相关 pod。应和创建时的 component label 做精确匹配，或保存完整 sanitized service_name label。

影响：这个问题合入后会在对应部署、运行或权限场景中留下真实故障/安全风险，后续排查成本较高。
建议：沿着上述风险点补齐校验、配置来源、权限边界或回归测试，避免同类问题再次出现。

**Review by Dallas98 — COMMENTED** [review]
事后审查补充：按已合入 diff 补充 5 条可行动 inline review comments。

File: sdk/nexent/container/k8s_client.py (hunk 1 above)
Comment (Dallas98) [inline sdk/nexent/container/k8s_client.py:28]: 二次事后审查补充：[P2] 注释写 max 253，但这个 sanitizer 没有实际 enforce 长度，而且很多使用点是 label value/名称片段，常见上限是 63。注释会让调用方误以为长度已经安全，建议把长度作为参数并在函数内截断。

影响：该问题合入后仍可能在真实部署、运行、权限或测试场景中形成回归风险。
建议：后续按这个风险点补齐边界校验、配置来源收敛、权限约束或针对性回归测试。

**Review by Dallas98 — COMMENTED** [review]
二次事后审查补充 1/5：单条独立 review submission。

File: sdk/nexent/container/k8s_client.py (hunk 5 above)
Comment (Dallas98) [inline sdk/nexent/container/k8s_client.py:53]: 二次事后审查补充：[P2] name 标注为 str，但测试又传 None；如果传入 truthy 的非字符串对象，name.lower() 会抛 AttributeError。边界入口应先 str(name) 或显式拒绝非字符串。

影响：该问题合入后仍可能在真实部署、运行、权限或测试场景中形成回归风险。
建议：后续按这个风险点补齐边界校验、配置来源收敛、权限约束或针对性回归测试。

**Review by Dallas98 — COMMENTED** [review]
二次事后审查补充 2/5：单条独立 review submission。

File: sdk/nexent/container/k8s_client.py (hunk 3 above)
Comment (Dallas98) [inline sdk/nexent/container/k8s_client.py:66]: 二次事后审查补充：[P1] 返回值没有任何长度限制，长 service_name 经过清洗后仍然可以非常长，最终 pod name 会超过 Kubernetes 限制。这里应该在 sanitizer 或 _generate_pod_name 中截断并附稳定 hash。

影响：该问题合入后仍可能在真实部署、运行、权限或测试场景中形成回归风险。
建议：后续按这个风险点补齐边界校验、配置来源收敛、权限约束或针对性回归测试。

**Review by Dallas98 — COMMENTED** [review]
二次事后审查补充 3/5：单条独立 review submission。

File: sdk/nexent/container/k8s_client.py (hunk 6 above)
Comment (Dallas98) [inline sdk/nexent/container/k8s_client.py:123]: 二次事后审查补充：[P2] tenant_part/user_part 只取前 8 位，tenantabcdef 和 tenantabcxyz 会生成同样前缀，排障和清理时容易误判。建议用前缀+短 hash，而不是纯截断。

影响：该问题合入后仍可能在真实部署、运行、权限或测试场景中形成回归风险。
建议：后续按这个风险点补齐边界校验、配置来源收敛、权限约束或针对性回归测试。

**Review by Dallas98 — COMMENTED** [review]
二次事后审查补充 4/5：单条独立 review submission。

File: sdk/nexent/container/k8s_client.py (hunk 7 above)
Comment (Dallas98) [inline sdk/nexent/container/k8s_client.py:531]: 二次事后审查补充：[P2] list_containers 用 sanitize 后的 safe_name 再做子串匹配，service=api 会匹配到 my-api-v2 等不相关 pod。应和创建时的 component label 做精确匹配，或保存完整 sanitized service_name label。

影响：该问题合入后仍可能在真实部署、运行、权限或测试场景中形成回归风险。
建议：后续按这个风险点补齐边界校验、配置来源收敛、权限约束或针对性回归测试。

**Review by Dallas98 — COMMENTED** [review]
二次事后审查补充 5/5：单条独立 review submission。
